### DISCUSSION TITLE
SVG export with shape ids

### DISCUSSION THREAD

**Discussion by PhbsSmn** [jgraph/drawio#2667] [Q&A] [answered]
Currently I noticed that the id of a shape can be found in Arrange -> Algin -> Edit data  and there on top you have then in this case: **npx8zoqcDDIV45s1C4aH-20**
![image](https://user-images.githubusercontent.com/26283753/156731472-02a7fbae-8fc3-4d1a-8c93-867dbe54f471.png)

Is there a possibility to export the id of a shape or set a custom one in drawio and export it to an SVG image?

**Comment by linghushaoxia** [comment, marked as answer]
please refer the iissue 77, the svgdata plugin works.
https://github.com/jgraph/drawio/issues/77

**Reply by geiseri** [reply]
Is there a way to add the id directly on the object that the id was set on vs creating a group wrapping it?  I was very confused why updating the object from my vue app never seemed to apply.🤣

**Reply by geiseri** [reply]
The issue with that one is that it radically changes the structure of the SVG enough to cause things like text boxes to be a nightmare to access via the DOM.

**Comment by RoyiAvital** [comment]
Is there a way to change the ID?
I want to easily select them later.

**Reply by dsample** [reply]
From the screenshot above, double clicking on the ID opens a dialog to change it.
